src/components/common/modal/Modal.jsx

```diff
@@ -0,0 +1,77 @@
+import { useEffect } from 'react';
+import Button from '@/components/common/button/Button';
+import ModalHeader from '@/components/common/modal/ModalHeader';
+import PostContent from '@/components/common/modal/PostContent';
+import ModalPortal from './ModalPortal';
+
+const Modal = ({
+  isOpen,
+  onClose,
+  contentHtml,
+  sender,
+  role,
+  profileImgUrl,
+  createdAt,
+  font,
+}) => {
+  useEffect(() => {
+    document.body.style.overflow = isOpen ? 'hidden' : '';
+    return () => (document.body.style.overflow = '');
+  }, [isOpen]);
+
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+    const handleKeyDown = (event) => {
+      if (event.key === 'Escape') {
+        onClose();
+      }
+    };
+    document.addEventListener('keydown', handleKeyDown);
+    return () => document.removeEventListener('keydown', handleKeyDown);
+  }, [isOpen, onClose]);
+
+  if (!isOpen) {
+    return null;
+  }
+
+  return (
+    <ModalPortal>
+      <div
+        className="fixed inset-0 z-[1000] bg-black/50"
+        onClick={onClose}
+        aria-hidden="true"
+      />
+
+      <div
+        className="tablet:w-[500px] mobile:w-[90vw] mobile:h-auto fixed left-1/2 top-1/2 z-[1001] flex h-[476px] max-h-[90vh] w-[600px] max-w-[90vw] -translate-x-1/2 -translate-y-1/2 flex-col overflow-hidden rounded-[16px] bg-white p-5 shadow-lg"
+        role="dialog"
+        aria-modal="true"
+        onClick={(e) => e.stopPropagation()}>
+        <ModalHeader
+          sender={sender}
+          role={role}
+          date={createdAt}
+          profileImgUrl={profileImgUrl}
+        />
+
+        <div className="flex-grow overflow-y-auto px-4 py-2">
+          <PostContent htmlContent={contentHtml} font={font} />
+        </div>
+
+        <div className="flex justify-center pt-6">
+          <Button
+            onClick={onClose}
+            theme="primary"
+            size={56}
+            className="font-pretendard flex h-[40px] w-[120px] items-center justify-center gap-[10px] rounded-[6px] border-0 bg-[#9935FF] text-[16px] font-bold leading-[26px] tracking-[-0.16px] text-white transition-all duration-150 hover:bg-[#8a2fe6] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[#7a29cc] active:bg-[#7a29cc] disabled:cursor-default disabled:bg-gray-300 disabled:text-white">
+            확인
+          </Button>
```

Comment:
size를 40으로 변경해주시면 h-[40px]을 안넣으셔도 됩니다!
지금 겹치는 스타일이 많습니다. 버튼 컴포넌트에 이미 정의된 스타일 속성도 있어서 아래처럼 바꿔주시면 될 것 같습니다.
focus, hover, disabled는 버튼 컴포넌트에 이미 정의되어있습니다!

```suggestion
<Button
    theme="primary"
    size={40}
    onClick={onClose}
    className="font-16-regular w-[120px]">
    확인
</Button>
```

<img width="141" height="61" alt="Image" src="https://github.com/user-attachments/assets/1d4e838b-c953-4af3-937c-85b4ce877e14" />